### DISCUSSION TITLE
Potentially reduced transcription performance compared to older version.

### DISCUSSION THREAD

**Discussion by zappa2** [rakuri255/UltraSinger#138] [General]
Hi Rakuri,

let me start to give compliments about this brilliant software! 
Works really well for songs with clear vocals and we have spent hours singing the Ultrasinger songs!!

When songs have less clear vocals, I started noticing big differences between the current version (Version: 0.0.9 Date: 2024.02.06) and an older version. In various songs (both EN and NL) It seems the transcription goes out of sync and keeps repeating words. 

Example of the (sorry, Dutch) song at this url:  [https://www.youtube.com/watch?v=z9E4Nl2d60U](url)
This is a very difficult song to transcribe because of the heavy processing, so some errors are to be expected. 
However, the transcription with the new version of Ultrasinger goes completely of the rails.   

Old version of Ultrasinger does a much better job:
[Linda, Roos & Jessica - Ademnood.txt](https://github.com/rakuri255/UltraSinger/files/15099550/Linda.Roos.Jessica.-.Ademnood.txt)

New version keeps "hanging" in the la la la la and ohohohoh: 
[Linda, Roos & Jessica - Ademnood NEW.txt](https://github.com/rakuri255/UltraSinger/files/15099597/Linda.Roos.Jessica.-.Ademnood.NEW.txt)

Command:
py UltraSinger.py -i https://www.youtube.com/watch?v=z9E4Nl2d60U --language nl --disable_karaoke True

Old version source:
[UltraSinger-main.zip](https://github.com/rakuri255/UltraSinger/files/15099625/UltraSinger-main.zip)

Old version console output:
[console_output_old.txt](https://github.com/rakuri255/UltraSinger/files/15099774/console_output_old.txt)

New version source:
[UltraSinger-main (1).zip](https://github.com/rakuri255/UltraSinger/files/15099629/UltraSinger-main.1.zip)

New version console output:
[console_output_new.txt](https://github.com/rakuri255/UltraSinger/files/15099778/console_output_new.txt)

What could be happening?  (please note I am testing the transcription with the old and new version right after eachther on the same system, with the same command.  System should be no problem, with Core i5 13600k, 64 GB RAM and RTX 4070 Ti Super 16 GB.

Thnx,
DJP!

**Comment by rakuri255** [comment]
Thanks for the info!
I lately changed the filters to reduce the noise in the vocal tracks. 
Maybe the settings are too aggressive for this track.

I'll test it when I have time :)

As for the hanging words, that's often a side effect of WhisperX. Sometimes there are halizinations. I have a few ideas to "reduce" them.

**Reply by zappa2** [reply]
Thanks for your quick response! 
Is there a parameter in the code or config file I could change to reduce the amount of noise filtering?  
That would enable me to do some tests.
Thnx!
